packages/core/test/taskExecutor.test.ts

```
@@ -0,0 +1,1716 @@
+    const executionOrder: string[] = [];
+    const expectedError = new Error("Middleware error");
+
+    // Register global middleware that throws an error
+    lifecycleHooks.registerGlobalMiddlewareHook({
+      id: "global-middleware",
+      fn: async ({ payload, ctx, next }) => {
+        executionOrder.push("middleware-before");
+        throw expectedError;
+        // Should never get here
+        await next();
+        executionOrder.push("middleware-after");
```

Biome (1.9.4)

[error] This code is unreachable

... because this statement will throw an exception beforehand


(lint/correctness/noUnreachable)